libraries/chain/abi_serializer.cpp

```diff
@@ -228,13 +237,35 @@ namespace eosio { namespace chain {
          return type;
    }
 
+   // true if std::string can be converted to name
+   bool abi_serializer::is_string_valid_name(std::string_view str) const
+   {
+      if (size_t len = str.size(); len > 13)
+         return false;
+
+      for( const char c : str ) {
+         if( c >= 'a' && c <= 'z' )
+            continue;
+         else if( c >= '1' && c <= '5' )
+            continue;
+         else if ( c == '.' )
+            continue;
+         else
+            return false;
+      }
+      return true;
+   }
```

Comment:
This should be moved to `name.hpp` and reuse existing code.

Reply:
moved to name.hpp